### PR TITLE
fix(VTreeview): select & activate issues

### LINKED ISSUE #19400: [Bug Report][3.5.9] Treeview: when selecting a checkbox of the treeview item then the v-model:selected not always updates the given property
### Environment
**Vuetify Version:** 3.5.9
**Vue Version:** 3.4.21
**Browsers:** Chrome 122.0.0.0
**OS:** Windows 10

### Steps to reproduce
Unfold the tree-view items... And also selected them.

### Expected Behavior
I'm expecting that every selected treeviewitem is shown in the list with selected treeview items.

### Actual Behavior
It seems that mostly the treeviewitems with childs are selected properly, but the one without childs never get selected (in the resultset). And sometimes, depending on if the treeview is folded, or not, then also that treeview is not placed in the result-set while they are selected

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNqVVlGP2zYM/itqMOAcIHaarSvQ4JLt1tdhDyv2dHdYFVuJ1ZMtQ5KTyw757yMl2ZZsF9gCtCeRFEV+/Ej58W2hVb5+aJrs3LLFdnFvWNUIatj+qSbk/pzmVBV2bXdGSnGgyu0JyaWQave0aBSvqLo+LTrFEVy4tT8LpzvP5JxqIc22UaxhdefcX8BzWRNtqDL7quCp5uLM1IUqdr92SjCPDviIUsONYPvfZU4FKaXRDfzDM7G+D2Yd5YmC3rS/AbwreRnOABhSxOEiOqlhryYQu7AUY2fOLqGYQOaVLJjYaiZYblgB0KHhgJv7bTlEp0Fp/060QtIizUsuCsVqsML92Ii9NrQuLGJggVDmJTsrWaeFvNRja14XzDBV8RoQCQ8dpHyByr6koGonkWB4Dlgwr2k1ScQanKlov2cgj8fZ62RrBK+n5vWc9diqQxfLZekJ6fGTnDgD9qXgMRc8f4lViplWgfLwDRzNOacHwUL53vLH1zykCIinHLFSS6WAWwU/84IpAnw3HFhsXXphbAlHhxtgg0zZ/BimVyGzPg6S/0RauGvMVn70DM0Eq0+mJLvdjrwfA/nCrmiGPBirckE1hoe3peVHcpS1SS+Mn0qTCvyfWM1JsStpaPrBbXNWAxtjX1GshHyxZSBX2SpypGepgGrkoNiFKc50lNcaEoumhgVB50oKkb5C1WitueEysiDkpGTbxKISaoGMEYye4/LHdohxyUeHEc+jpWPiOAQ3rghfQvORuSEAje6A5TMahAlUvaOpSYd8RdPNjNZ3BgI/1frJHPUaGxfENgP/BztbV1SIsdbyF2GIWQbCGej/R8t0Ozea+0nddcpc13ScpxYrHU70gwnK7lH9k2lmwnTOVHFaowotQs2vdnj4JiE78vgcNB2GAv7j5tUNzX2Yfhmpo3hoMy4DrAw0R6qRgEEcQT1hqBUwFOFPVHef25fRwSE1+2TPBR8VJYSwE8Imek9hCyXmjasevERSGVKwI22FIW/OYUEN3ZJkSXZ7kngZGRp4C0iuOiliGwuuTWRyW8LCLXNZNS3M/m13kXuENF7VS9Cs1oZ0bygUzpRcZ9ZvVtEmwVUcmvUEblHTB4I/fNZAjOdPzPwBO3saQwr6zd80GH72konxbbkMh5V7isjjJJDNTBR3D0KQ3zoQ72ZDCK967ta3DskeyAs1eRmg2JeGJPCgR2AO2AGQoMwUK9qcJQnN8xXJW2WrHCXg8Hcoo0Xm3F//RlE8q/mRJO/AU8brXLQF0w6xJUFZ0+rS7eNDHjUwCbJdAWOWmQY2JmD+vcQrZkpZAL36gPH7asQfDMqm3cPiX8ilv3qmhEcGiCZ3pTGN3q7XtOEZfoL4zItDJtVp3fu760PEX2ZKoIpChPfgT2fftKwTZMrYBhvLMjcOD2BGzehEjkVOmFJ4AmsiBTz18oSiwbTDCN7EgbdwAcI+barwShiHAQ7w+pEkNLrC959jT38ocuhw9rYRRcg7+BBxAeBIhI/TEWuGuliKRE4h8azzFrXIbDN316MqbupbAOZtpuBDDJZzCV2Rw7QVOqraC8ieHNziF7IhWwIzPBgMswXBIEliowsde7dff3hDVZaXVD2Y5P0yM/IveFbUZ6pZsrx5tYZnjCWb5e3rXGPY9O7XfqgvbqvFT9mHbPNpgYufs0+L538BRH8ijg==)

### LINKED ISSUE #19402: [Bug Report][3.5.9] Treeview: when folding/unfolding a selectable treeview item, then the checkbox should not be checked
### Environment
**Vuetify Version:** 3.5.9
**Vue Version:** 3.4.21
**Browsers:** Chrome 122.0.0.0
**OS:** Windows 10

### Steps to reproduce
Unfold a treeview-item => checkbox is checked

### Expected Behavior
I'm expecting that a checkbox is not automatically checked when folding/unfolding the treeview

### Actual Behavior
The checkbox of the unfolded treeview-item is checked

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNqVVlGP2zYM/itqMOAcIHaarSvQ4JLt1tdhDyv2dHdYFVuJ1ZMtQ5KTyw757yMl2ZZsF9gCtCeRFEV+/Ej58W2hVb5+aJrs3LLFdnFvWNUIatj+qSbk/pzmVBV2bXdGSnGgyu0JyaWQave0aBSvqLo+LTrFEVy4tT8LpzvP5JxqIc22UaxhdefcX8BzWRNtqDL7quCp5uLM1IUqdr92SjCPDviIUsONYPvfZU4FKaXRDfzDM7G+D2Yd5YmC3rS/AbwreRnOABhSxOEiOqlhryYQu7AUY2fOLqGYQOaVLJjYaiZYblgB0KHhgJv7bTlEp0Fp/060QtIizUsuCsVqsML92Ii9NrQuLGJggVDmJTsrWaeFvNRja14XzDBV8RoQCQ8dpHyByr6koGonkWB4Dlgwr2k1ScQanKlov2cgj8fZ62RrBK+n5vWc9diqQxfLZekJ6fGTnDgD9qXgMRc8f4lViplWgfLwDRzNOacHwUL53vLH1zykCIinHLFSS6WAWwU/84IpAnw3HFhsXXphbAlHhxtgg0zZ/BimVyGzPg6S/0RauGvMVn70DM0Eq0+mJLvdjrwfA/nCrmiGPBirckE1hoe3peVHcpS1SS+Mn0qTCvyfWM1JsStpaPrBbXNWAxtjX1GshHyxZSBX2SpypGepgGrkoNiFKc50lNcaEoumhgVB50oKkb5C1WitueEysiDkpGTbxKISaoGMEYye4/LHdohxyUeHEc+jpWPiOAQ3rghfQvORuSEAje6A5TMahAlUvaOpSYd8RdPNjNZ3BgI/1frJHPUaGxfENgP/BztbV1SIsdbyF2GIWQbCGej/R8t0Ozea+0nddcpc13ScpxYrHU70gwnK7lH9k2lmwnTOVHFaowotQs2vdnj4JiE78vgcNB2GAv7j5tUNzX2Yfhmpo3hoMy4DrAw0R6qRgEEcQT1hqBUwFOFPVHef25fRwSE1+2TPBR8VJYSwE8Imek9hCyXmjasevERSGVKwI22FIW/OYUEN3ZJkSXZ7kngZGRp4C0iuOiliGwuuTWRyW8LCLXNZNS3M/m13kXuENF7VS9Cs1oZ0bygUzpRcZ9ZvVtEmwVUcmvUEblHTB4I/fNZAjOdPzPwBO3saQwr6zd80GH72konxbbkMh5V7isjjJJDNTBR3D0KQ3zoQ72ZDCK967ta3DskeyAs1eRmg2JeGJPCgR2AO2AGQoMwUK9qcJQnN8xXJW2WrHCXg8Hcoo0Xm3F//RlE8q/mRJO/AU8brXLQF0w6xJUFZ0+rS7eNDHjUwCbJdAWOWmQY2JmD+vcQrZkpZAL36gPH7asQfDMqm3cPiX8ilv3qmhEcGiCZ3pTGN3q7XtOEZfoL4zItDJtVp3fu760PEX2ZKoIpChPfgT2fftKwTZMrYBhvLMjcOD2BGzehEjkVOmFJ4AmsiBTz18oSiwbTDCN7EgbdwAcI+barwShiHAQ7w+pEkNLrC959jT38ocuhw9rYRRcg7+BBxAeBIhI/TEWuGuliKRE4h8azzFrXIbDN316MqbupbAOZtpuBDDJZzCV2Rw7QVOqraC8ieHNziF7IhWwIzPBgMswXBIEliowsde7dff3hDVZaXVD2Y5P0yM/IveFbUZ6pZsrx5tYZnjCWb5e3rXGPY9O7XfqgvbqvFT9mHbPNpgYufs0+L538BRH8ijg==)

### LINKED ISSUE #19441: [Documentation] VTreeview expand nodes on click and parent nodes cannot be activatable
### Environment
**Vuetify Version:** 3.5.10

**Browsers:** Chrome 122.0.0.0
**OS:** Windows 10

### Steps to reproduce
Click on any nodes in the basic activatable example

### Expected Behavior
- To work like the [Vuetify 2 example](https://v2.vuetifyjs.com/en/components/treeview/#activatable)
  - Node expand only with icon click
  - Node can be activated

### Actual Behavior
- Node expand on click anywhere in the line
- Node cannot be activated

### Reproduction Link
[https://vuetifyjs.com/en/components/treeview/#activatable](https://vuetifyjs.com/en/components/treeview/#activatable)

### LINKED ISSUE #19471: [Bug Report][3.5.11] VTreeview selection not working
### Environment
**Vuetify Version:** 3.5.11
**Vue Version:** 3.4.21
**Browsers:** Chrome 123.0.0.0
**OS:** Mac OS 10.15.7

### Steps to reproduce
See the playground

### Expected Behavior
Selected items should be exposed in v-model or in v-model:selected

### Actual Behavior
v-model and v-model:selected stays an empty array

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNp9VMtu2zAQ/JUte3ACWDIcJwEqOAaKHnrroe3N9oGW1g0TmhQoWmkg6N9LUg+SthoebHF2dji7XGnbkK9lmdZnJBlZazyVnGrc7ATAuk5yKTRlApUDHFQhx1x3W4A6OckC+dOOdDiT4vd7iTsyEDJmNCsT38440uNsDjMmCizR/Ag923smpwcn9GsQAh0obdaL4WzvRcm3ftO75ePWAVoh1gzfPDjp2HuIHLv/i9iYk1hz/y/bLmmKTCjnIaZQn5VI5OElaKJdnQ49cPSoq3mowRdqwLBSU2fBalagghqVZjnlLrEHLxrk1XNOK1tmSZPH0LghWTjAwqYOE2L6yI6G9mlsQMpR/NHPOxLQAX5IEKbfVV8gFmkgtojm7foE5FUYMkFTVLi313mUyviwhwAT/TFXl2qu7hXfe17KijganQHQNM5zKugJoW0jAwvjIHR7XUJ8P3Y3jGkX8W9UlGy2Va5Y2Y03/i2l0lDgkZ65hqYTK6imGdzcwtMGbnpsGJ1+BDPo37OraAbb/Yi62TaIr2SUc+Eig+VItkszza34Tym1F3fz8sx4odDqX7TRydzNx9xvlgmflzNoI4WBurqi3k1Q423n9f6C5P32QqvY84e+vaEHb+i7oqJwKZMF+JzH6ZyJSgD8fXQrZoRRHxnQ9tY9mOlcL/rBIXPCmXit0pdKCvM5b3Ykr8wXLNvuWxNjJztTyYmWAaEDLalpW9LOySq9T5dfDH2VPqTLJZlrdcb9P0gds24=)

### Other comments
All features that are dependent on selection do not work.

### LINKED ISSUE #19533: [Bug Report][3.5.13] VTreeView Select Functionality (Prepend slot, selectable prop v-model value, click:select event)
### Environment
**Vuetify Version:** 3.5.13
**Last working version:** 2.7.1
**Vue Version:** 3.4.21
**Browsers:** Chrome 123.0.0.0
**OS:** Windows 10

### Steps to reproduce
- Use a prepend slot in conjunction with the selectable prop on the tree

- Utilize the "selectable" prop to populate the v-model array with selected values

- Trigger a method when a selectable checkbox is selected

### Expected Behavior
- Checkbox shows up next to prepend icon 

- Selecting the "selectable" checkbox adds the value to the v-model array

- Event click:select is triggered when checkbox is selected

### Actual Behavior
- Prepend slot overwrites the checkbox when the "Selectable" prop is selected

- Event click:select does not trigger when selecting the built-in checkbox

- v-model array for treeview does not update when selectable checkboxes are selected

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNqtVl1v2jAU/SsWLwGpCU97iShbpfVxmrTXUk3GuQkujh0lTkqF8t93bZPEpjCGtD6Afc69x/fLpi/HWVOz5VNVJV0Ls3S20lBWgmpYbyQhx6OuAfoeF7SqBDR9b+BVF2+1JN+Y4Gz/uJlpaPQ8ograaLGZrc1itbQ2VgXNjUzH4d1sCeniUmUgjCfim5lDU45nNwja7xFVFUjIDCy55lSIj5+IDLSxjffwgbyk5ahFmeYd1XQrwAFGJVYytiE7qAEBzDNx6aQOPmXl9GwWmMdQG0ygEUqnVQ0oa2I72kAe7DGkxxo4D5s7Z0qiB89PqSVsx0VWmxxGM1Nq5/yVRGXG41yJDOrYQBFJfSwirglWHats5C8cB6KBNbFuXECcKdaWIHWsWi24hHPX1TJovGGHniEQsLhtWM0rbS3hUKlakwxy2gpNjk6uBL1TWZMOe0JMNVOStxI7gwHOTSUWE40GO94kbsqSqm12zmLg+we3Gr4zbG5K5gvyuCbzUSUYkZS8RFW7xaZGrycvQkwx/LAI2elSnAosqCxaWkBssC/R6ETIG/pYE4mD+9aElMKjLMmQjM3e58vsXL2k9T5T74FVleVDl027cBtv1SGwkIVvwUuU8nl9wPpO/HnDfdODGNKxpnBgICZ+KDFK4gRgFafqufYEkL2siHiTPC1RgmuBGlFScO2HMB1yxcEU+jc+Ey0e6DuegkAL+94ETEdFaxim2F1nnaYk0BouaZDaJwVfpdFUn6v8Vemimq8oVKES7PtnzWGUTfCXeT9p9zf17LLF1cRy2pmXIsGPzyddj+Kf9bnM4JCYO3dV/hIZ6vvZ3Wy4mUZeSFUH12I8Da/SXfOzpVsQCZYo50USvg6jZgjfHknK9njBUS18TTy9kLip+Ov56fuP56TMLsqF8E0x/Gfh/+b7QWuZCMX29zVk6Hq/sAv8YVwtxx+nWf8gWyHsx+sfluhydg==)

### Other comments
documentation on how to use v-treeview-item would be nice

### PR DESCRIPTION
**Pull request by yuwu9145**
Properly hook nested  composable into treeview

fixes #19441
fixes #19402
fixes #19400
fixes #19533
fixes #19471

vuetify-treeview-demo

## Markup:

```vue
<template>
  <h2>Activatable (independent)</h2>
  <v-treeview
    :items="items"
    active-strategy="independent"
    item-title="title"
    item-value="id"
    activatable
    @update:activated="activateSelectCallback"
  />
  <h2>Activatable (single-independent)</h2>
  <v-treeview
    :items="items"
    active-strategy="single-independent"
    item-title="title"
    item-value="id"
    activatable
    @update:activated="activateSelectCallback"
  />
  <h2>Activatable (single-leaf)</h2>
  <v-treeview
    :items="items"
    active-strategy="single-leaf"
    item-title="title"
    item-value="id"
    activatable
    @update:activated="activateSelectCallback"
  />
  <h2>Activatable (leaf)</h2>
  <v-treeview
    :items="items"
    active-strategy="leaf"
    item-title="title"
    item-value="id"
    activatable
    @update:activated="activateSelectCallback"
  />
  <h2>Activatable with open-on-click</h2>
  <!-- <v-treeview
    :items="items"
    activatable
    activate-strategy="independent"
    open-on-click
    @update:activated="activateSelectCallback"
  /> -->
  <h2>Selectable (single-leaf)</h2>
  <v-treeview
    :items="items"
    item-title="title"
    item-value="id"
    select-strategy="single-leaf"
    selectable
    @update:selected="activateSelectCallback"
  />
  <h2>Selectable (leaf)</h2>
  <v-treeview
    :items="items"
    item-title="title"
    item-value="id"
    select-strategy="leaf"
    selectable
    @update:selected="activateSelectCallback"
  />
  <h2>Selectable (independent)</h2>
  <v-treeview
    :items="items"
    item-title="title"
    item-value="id"
    select-strategy="independent"
    selectable
    @update:selected="activateSelectCallback"
  />
  <h2>Selectable (single-independent)</h2>
  <v-treeview
    :items="items"
    item-title="title"
    item-value="id"
    select-strategy="single-independent"
    selectable
    @update:selected="activateSelectCallback"
  />
  <h2>Selectable (classic)</h2>
  <v-treeview
    :items="items"
    item-title="title"
    item-value="id"
    select-strategy="classic"
    selectable
    @update:selected="activateSelectCallback"
  />
  <h2>No Activatable</h2>
  <!-- <v-treeview
    :items="items"
    @update:activated="activateSelectCallback"
  /> -->
</template>

<script setup>
  import { ref } from 'vue'

  function activateSelectCallback ( e ) {
    console.log('activateCallback', e)
  }
  const items = ref([
    {
      id: 1,
      title: 'Applications :',
      children: [
        { id: 2, title: 'Calendar : app' },
        { id: 3, title: 'Chrome : app' },
        { id: 4, title: 'Webstorm : app' },
      ],
    },
    {
      id: 5,
      title: 'Documents :',
      children: [
        {
          id: 6,
          title: 'vuetify :',
          children: [
            {
              id: 7,
              title: 'src :',
              children: [
                { id: 8, title: 'index : ts' },
                { id: 9, title: 'bootstrap : ts' },
              ],
            },
          ],
        },
        {
          id: 10,
          title: 'material2 :',
          children: [
            {
              id: 11,
              title: 'src :',
              children: [
                { id: 12, title: 'v-btn : ts' },
                { id: 13, title: 'v-card : ts' },
                { id: 14, title: 'v-window : ts' },
              ],
            },
          ],
        },
      ],
    },
    {
      id: 15,
      title: 'Downloads :',
      children: [
        { id: 16, title: 'October : pdf' },
        { id: 17, title: 'November : pdf' },
        { id: 18, title: 'Tutorial : html' },
      ],
    },
    {
      id: 19,
      title: 'Videos :',
      children: [
        {
          id: 20,
          title: 'Tutorials :',
          children: [
            { id: 21, title: 'Basic layouts : mp4' },
            { id: 22, title: 'Advanced techniques : mp4' },
            { id: 23, title: 'All about app : dir' },
          ],
        },
        { id: 24, title: 'Intro : mov' },
        { id: 25, title: 'Conference introduction : avi' },
      ],
    },
  ])
</script>
```

### CODE REVIEW

File: packages/vuetify/src/labs/VTreeview/VTreeviewItem.tsx [hunk 1]
```diff
@@ -33,34 +36,133 @@ export const VTreeviewItem = genericComponent<VListItemSlots>()({
...
     const visibleIds = inject(VTreeviewSymbol, { visibleIds: ref() }).visibleIds
 
     useRender(() => {
+      const hasTitle = (slots.title || props.title != null)
+      const hasSubtitle = (slots.subtitle || props.subtitle != null)
       const listItemProps = VListItem.filterProps(props)
       const hasPrepend = slots.prepend || props.toggleIcon
 
-      return (
+      return isActivetableGroupActivator.value
```
Comment (yuwu9145) [inline packages/vuetify/src/labs/VTreeview/VTreeviewItem.tsx:103]: Reason for making custom activetable group activator is VList Group doesn't allow group activator to be activated.

#19479

Comment (KaelWD) [reply packages/vuetify/src/labs/VTreeview/VTreeviewItem.tsx:103]: activatable not activetable

File: packages/vuetify/src/labs/VTreeview/VTreeviewChildren.tsx [hunk 2]
```diff
@@ -60,29 +62,33 @@ export const VTreeviewChildren = genericComponent<new <T extends InternalListIte
...
-              />
-            )
-            : undefined,
+        prepend: slotProps => (
+          <>
+            { props.selectable && (!children || (children && !['leaf', 'single-leaf'].includes(props.selectStrategy as string))) && (
+              <div>
+                <VCheckboxBtn
+                  key={ item.value }
+                  tabindex="-1"
```
Comment (johnleider) [inline packages/vuetify/src/labs/VTreeview/VTreeviewChildren.tsx:80]: This has mixed results. When the parent is expandable and selectable, there's not a way to do so with keyboard only.

Comment (yuwu9145) [reply packages/vuetify/src/labs/VTreeview/VTreeviewChildren.tsx]: fixed

**Comment by yuwu9145** [comment]
This PR is ready for a further check/review.

There is a known issue of `return-object` doesn't work, I found the solution but needs more time to verify/test, so it will be addressed in a separate later.

This PR's focus is only on select/activate

**Comment by johnleider** [comment]
Looking good. The only thing I've come across is active applying when closing a parent.
tree

File: packages/docs/src/pages/en/styles/flex.md [hunk 3]
```diff
@@ -52,21 +52,41 @@ Control the layout of flex containers with alignment, justification and more wit
 | **flex-md-0-0** | flex: 0 0 auto; |
```
Comment (yuwu9145) [inline packages/docs/src/pages/en/styles/flex.md:52]: Is this supposed to be in this PR?

**Comment by yuwu9145** [comment]
> The only thing I've come across is active applying when closing a parent.

Fixed

**Pull request merged**